### PR TITLE
Add back the otlpmetricgrpc exporter

### PR DESCRIPTION
**Pull request by MrAlias**
Part of #2807

Notable differences from the `main` branch exporter:

- `NewUnstarted` is removed. The `otlpmetric.Client` no longer has a concept of started or unstarted, the new exporter returned is always assumed to be ready to export.
- `NewClient` is removed. Use `otlpmetricgrpc.New` to create a usable `metric.Exporter`.
- The client does not handle concurrency requirements, that is a concern of `otlpmetric.Exporter`.
- The client does not handle shutdown once requirements, that is a concern of the `otlpmetric.Exporter`.

This PR does not include full integration testing. I plan to address that in a follow-on PR that is being tracked in https://github.com/open-telemetry/opentelemetry-go/issues/2807.

### CODE REVIEW

File: exporters/otlp/otlpmetric/otlpmetricgrpc/client.go [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+func (c *client) Shutdown(ctx context.Context) error {
+	// The otlpmetric.Exporter synchronizes access to client methods and
+	// ensures this is called only once. The only thing that needs to be done
+	// here is to release any computational resources the client holds.
+
+	c.metadata = nil
+	c.requestFunc = nil
+	c.msc = nil
+
+	err := ctx.Err()
```
Comment (ash2k) [inline exporters/otlp/otlpmetric/otlpmetricgrpc/client.go:118]: gRPC `Close()` is probably always instantaneous. It'd be better to just always call it to clean up the connection (if `ourConn` is true).

Comment (MrAlias) [reply exporters/otlp/otlpmetric/otlpmetricgrpc/client.go:118]: > gRPC `Close()` is probably always instantaneous.

Is it? It acquires a lock, and waits for the tear-down of all underlying connections:

https://github.com/grpc/grpc-go/blob/d5dee5fdbdeb52f6ea10b37b2cc7ce37814642d7/clientconn.go#L1014-L1062

If the user has included a deadline to this call, making the additional call to Close beyond the deadline would be counter to how the context is defined:

> work done on behalf of this context should be canceled

Comment (ash2k) [reply exporters/otlp/otlpmetric/otlpmetricgrpc/client.go:118]: Well, leaking the underlying gRPC connection with all the resources is not better than blocking. 🤷 It does lock a mutex, what I wanted to say is that it probably don't block for a significant duration i.e. presumably it does an abrupt connection close for all underlying streams and TCP connections and doesn't wait for the application code using them to handle that. I'm sorry, I don't have time now to dig into this properly, but not closing the gRPC conn just doesn't feel like the right thing to do. E.g. it's a typical pattern to `defer` the close then use a context. You don't care if the context was cancelled, you want the connection to be torn down anyway.

Comment (ash2k) [reply exporters/otlp/otlpmetric/otlpmetricgrpc/client.go:118]: > work done on behalf of this context should be canceled

I'm pretty sure that don't imply application should leak resources when a context is cancelled :)

**Review by jmacd — APPROVED** [review]

**Review by MadVikingGod — APPROVED** [review]
LGTM. 
Just a heads up you need 1 more review because jmacd and I both work for lightstep.

**Review by dmathieu — APPROVED** [review]

**Pull request merged**